evm/src/cpu/kernel/interpreter.rs

```diff
@@ -682,12 +749,42 @@ impl<'a> Interpreter<'a> {
     }
 
     fn run_opcode(&mut self) -> Result<(), ProgramError> {
+        // Jumpdest analysis is performed natively by the interpreter and not
+        // using the non-deterministic Kernel assembly code.
+        if self.is_kernel()
+            && self.generation_state.registers.program_counter
+                == KERNEL.global_labels["jumpdest_analysis"]
+        {
+            self.generation_state.registers.program_counter =
+                KERNEL.global_labels["jumpdest_analysis_end"];
+            self.generation_state
+                .set_jumpdest_bits(&self.generation_state.get_current_code()?);
+        }
+
         let opcode = self
             .code()
             .get(self.generation_state.registers.program_counter)
             .byte(0);
         self.opcode_count[opcode as usize] += 1;
         self.incr(1);
+
+        let op = decode(self.generation_state.registers, opcode)?;
+        self.generation_state.registers.gas_used += gas_to_charge(op);
+
+        #[cfg(test)]
+        if !self.is_kernel() {
+            println!(
+                "User instruction {:?}, stack = {:?}, ctx = {}",
+                op,
+                {
+                    let mut stack = self.stack();
+                    stack.reverse();
+                    stack
+                },
+                self.generation_state.registers.context
+            );
+        }
```

Comment:
Not a fan of this, as discussed elsewhere with @Nashtare I think this should be `log::trace!`

Reply:
Ok, but should I change it in this PR. I'm not sure if the interpreter is printing debug info in other places

Reply:
Yeah as this is now part of the prover logs (through CPU simulations), replacing the `println` by `log::trace` sounds better.

Reply:
I assume `#[cfg(debug_assertions)]` is also ok.